src/sortBy.ts

```diff
@@ -32,17 +60,51 @@ export function sortBy<T>(array: readonly T[], fn: (item: T) => any): T[];
  * @data_last
  * @category Array
  */
-export function sortBy<T>(fn: (item: T) => any): (array: readonly T[]) => T[];
+export function sortBy<T>(
+  fnDir: SortProjection<T> | Direction,
+  ...fnDirs: (SortProjection<T> | Direction)[]
+): (array: readonly T[]) => T[];
 
 export function sortBy() {
-  return purry(_sortBy, arguments);
+  if (Array.isArray(arguments[0])) {
```

Comment:
@TkDodo if I switch the signature to use tuples, this test will fail. I confess I don't fully understand purry. Do you have any suggestions about how I can make this test work if the rest args are also array types?

Reply:
which tests will fail? I suggest you commit it anyways, and then I can take a look at the failing tests.

Reply:
I think I have something reasonable now :)